### ISSUE TITLE
Add support for an ignore or keep argument.

### ISSUE THREAD

**Issue by rbscott** [apex/apex#825]
If you have a bug report please let us know the following:

  - Mac OS X 10.12.6:
  - Apex 0.15.0:

When appropriate please also paste the debug log output using `-ldebug` or `--log-level debug`.

Hello, there might be another way to handle this, but from my preliminary searching, I couldn't figure out the correct way to handle this case. I propose adding field to the function configuration file that instructs Apex to ignore certain parameters.

Currently, I am using Terraform to manage infrastructure and Apex to handle deployments. This is mostly OK, but unless you are careful, the two can step on one another. For instance, if you configure a Lambda function with VPC in Terraform, you have to copy all of the configuration values into Apex, otherwise Apex will overwrite all of the configuration from Terraform.

For instance:

1. Set up a SNS Queue, VPC and Lambda Function integrates into the VPC in Terraform.
1. Run `terraform apply`.
1. Deploy the function using Apex.
   1. Here Apex will overwrite the values for the VPC Configuration unless you copy over the new configuration into the function configuration file.

I use the two tools because (in my opinion) they are best suited for different purposes. Terraform is great at setting up infrastructure, but just OK at deployment. Apex is great at deployment, but isn't well suited for setting up infrastructure.

Since that is the case, would it make sense to add a flag such as `ignore`, `keep`, or `managed` to the configuration for a function?

For example:

```
{
    "role": "arn:aws:iam::xxxxx:role/special-role",
    "description": "Do something in Lambda",
    "handler": "main.lambda_handler",
    "environment": {
        "ENV_1": "VALUE"
    },
    "ignore": ["vpc", "deadletter_arn", "description"]
}
```

With this syntax, Apex would ignore whatever values are set for the VPC, Deadletter Queues and description. When creating a new function (It assumes another tool hasn't run), and it should be safe to use any provided values or defaults. This has the added benefit that it would also work with other infrastructure management tool like Cloud Formation.

Curious what people think. I would be happy to put together a PR if people agree this would be a useful feature.

**Comment by tsugitta** [comment]
I completely agree. In my case, I don't want to put `role` either because it's set by Terraform. but it's required for now.
